addons/binding/org.openhab.binding.toon/src/main/java/org/openhab/binding/toon/internal/ToonApiClient.java

```diff
@@ -73,10 +74,10 @@ public ToonApiClient(ToonBridgeConfiguration configuration) {
     public void login() throws ToonConnectionException {
         logger.debug("login start");
 
-        if (configuration == null || configuration.username == null || configuration.username.length() == 0) {
```

Comment:
configuration == null was always false here, while working on it I thought this made more sense, if needed I can change it.